### PR TITLE
Define is_zero for im and re

### LINKED ISSUE #14978: imaginary part of nonreal symbol cannot be zero
Start with `c = Symbol('c', real=False)`.

Good:
```
>>> c.is_zero
False
```

Room for improvement:
```
>>> im(c).is_zero          # gives None
```

A consequence of this is this Eq cannot simplify to False:
```
>>> Eq(im(c), 0)
Eq(im(c), 0)
>>> _.doit()
Eq(im(c), 0)
>>> simplify(_)
Eq(im(c), 0)
```

### PR DESCRIPTION
**Pull request by asmeurer**
Fixes #14978.

#### Release Notes

* functions
  - implemented `is_zero` for `im` and `re`

### CODE REVIEW

**Comment by asmeurer** [comment]
This time no fewer than 9 of the 34 builds had timed out network requests. I hope Travis is fixing this.

File: sympy/functions/elementary/tests/test_complexes.py [hunk 1]
```diff
@@ -892,3 +892,18 @@ def test_issue_14238():
     # doesn't cause recursion error
     r = Symbol('r', real=True)
     assert Abs(r + Piecewise((0, r > 0), (1 - r, True)))
+
+def test_zero_assumptions():
```
Comment (cbm755) [inline sympy/functions/elementary/tests/test_complexes.py]: local style is two blank lines

File: sympy/functions/elementary/tests/test_complexes.py [hunk 2]
```diff
@@ -892,3 +892,18 @@ def test_issue_14238():
...
+    nr = Symbol('nonreal', real=False)
+    ni = Symbol('nonimaginary', imaginary=False)
+    # imaginary implies not zero
```
Comment (cbm755) [inline sympy/functions/elementary/tests/test_complexes.py]: I don't find this comment useful here

File: sympy/functions/elementary/tests/test_complexes.py [hunk 3]
```diff
@@ -892,3 +892,18 @@ def test_issue_14238():
...
+    nzni = Symbol('nonzerononimaginary', zero=False, imaginary=False)
```
Comment (cbm755) [inline sympy/functions/elementary/tests/test_complexes.py]: maybe a more helpful variable name is `notonimaginaryaxis` but this is just nitpicking.

**Review by cbm755 — COMMENTED** [review]
LGTM.  Some optional comments as you wish.

File: sympy/functions/elementary/tests/test_complexes.py (hunk 2 above)
Comment (asmeurer) [reply sympy/functions/elementary/tests/test_complexes.py]: This is something we've wanted to change for some time (see https://github.com/sympy/sympy/issues/7649 and linked discussions). I think it's quite unintuitive that `imaginary` doesn't include zero, so I noted it here to indicate why I am testing both cases.

**Pull request merged**